tools/TestFx/TestEnvironmentFactory.cs

```diff
@@ -30,7 +30,7 @@ static TestEnvironmentFactory()
 
         internal static TestEnvironment BuildTestFxEnvironment()
         {
-            _environment.SetRecordedEnvironmentVariables();
+            _environment.SetEnvironmentVariables();
```

Comment:
why the files under TestFx need to be changed?

Reply:
For batch tests, the TestEnvironment object is initiated only once. However, the environment variables (set by SetEnvironmentVariables()) need to be re-populated for each test case. Previously, only Playback mode got it re-populated, so it caused SubscriptionId missing in json file during recording.

Reply:
Thanks @vidai-msft for fixing this.